Batteries/Data/Char/Basic.lean

```diff
@@ -22,3 +26,98 @@ theorem toNat_ofNat (n : Nat) : toNat (ofNat n) = if n.isValidChar then n else 0
   split
   · simp [ofNat, *]
   · simp [ofNat, toNat, *]
+
+/-- Returns `true` if `p` returns true for every `Char`. -/
+protected def all (p : Char → Bool) : Bool :=
+  Nat.all 1114112 fun c h =>
```

Comment:
I'm really not keen on all these unexplained magic constants. Can they be factored out? Linked to documentation?